### ISSUE TITLE
'Delete of an unqualified identifier in strict mode' introduced in 0.3.7

### ISSUE THREAD

**Issue by petemill** [hoxton-one/deepstream.io-client-js#59]
This issue was introduced in 0.3.7 (does not error in 0.3.6).

When passing `--use-strict` to node, the following line:
https://github.com/hoxton-one/deepstream.io-client-js/blob/master/src/utils/ack-timeout-registry.js#L62 

generates the following error:

```
x/node_modules/deepstream.io-client-js/src/utils/ack-timeout-registry.js:62
                delete timeout;
                       ^^^^^^^

SyntaxError: Delete of an unqualified identifier in strict mode.
```

I believe it would be a shame to leave this in since, prior to this addition, this module was perfectly compatible with strict-mode.

/cc @yasserf as it's your line addition

Thanks

**Comment by yasserf** [comment]
Many thanks for raising this, will fix it asap

Thanks

**Comment by WolframHempel** [comment]
Just released 0.3.8 that removes the delete statement. timeout was a local variable, so doesn't need explicit deletion. @petemill thanks for raising the issue

**Comment by yasserf** [comment]
This solution still doesn't delete the timeout from the map which means getting an ack response that was unexpected will still not throw an unsolicitated message.

Will raise a low priority issue to buff out tests around it and make sure it works -> #60
